[OSDOCS#18857]: Added a new module for Fencing Validator script (#110148)
openshift/openshift-docs: modules/Installation-fencing-validator-non-disruptive-checks.adoc

:_mod-docs-content-type: CONCEPT
[id="installation-fencing-validator-non-disruptive-checks_{context}"]

> = Run fencing validator script for non disruptive checks

Reviewer: ```suggestion
= Run fencing validator script for non-disruptive checks
```

Reviewer: A compound modifier in IBM is hyphenated.

[role="_abstract"]

> The simplest way to run the script is from a debug session on either control-plane node:

Reviewer: ```suggestion
The simplest way to run the script is from a debug session on either control-plane node.
```

[source,terminal]
----
$ oc debug node/\<node-name\> --chroot /host /usr/local/bin/fencing_validator
----

> This command does not reboot or fence any nodes. These checks are read-only and safe to run at any time and runs the following non-disruptive checks and reports the results:

Reviewer: ```suggestion
This command does not reboot or fence any nodes. These checks are read-only and safe to run at any time. They run and report the following non-disruptive checks:
```

. **OpenShift version check** - Confirms the cluster is running {product-title} 4.20.0 or later.
. **Node count check** - Confirms exactly 2 control-plane nodes exist.
. **Transport connectivity** - Establishes a connection to both nodes (via SSH or `oc debug`).
. **STONITH device check** - Verifies that STONITH devices are present and enabled in Pacemaker.
. **Pacemaker status** - Confirms both nodes are reporting ONLINE in the Pacemaker cluster.
. **Daemon health** - Checks that `corosync`, `pacemaker`, and `pcsd` services are active on both nodes.
. **etcd quorum** - Verifies that etcd has 2 healthy voting members and the cluster has quorum.
. **Fencing secrets** - Confirms that the fencing credential secrets (used by STONITH to authenticate to the BMC/IPMI) exist and are correctly bound to each node.

When all non-disruptive checks pass, the output looks something as follows:

[INFO]
====

> 

Reviewer: extra space

:_mod-docs-content-type: CONCEPT

> [id="installation-fencing-validator-non-disruptive-checks_{context}"]

Reviewer 2: Ensure filename and id match so xref linking is easier.

**I**nstallation-fencing-validator-non-disruptive-checks

Lowercase the "I". Ensure other files are the same.

. **STONITH device check** - Verifies that STONITH devices are present and enabled in Pacemaker.
. **Pacemaker status** - Confirms both nodes are reporting ONLINE in the Pacemaker cluster.
. **Daemon health** - Checks that `corosync`, `pacemaker`, and `pcsd` services are active on both nodes.
. **etcd quorum** - Verifies that etcd has 2 healthy voting members and the cluster has quorum.
. **Fencing secrets** - Confirms that the fencing credential secrets (used by STONITH to authenticate to the BMC/IPMI) exist and are correctly bound to each node.

When all non-disruptive checks pass, the output looks something as follows:

[source,terminal]
----

[INFO]
====
 OpenShift version: 4.20.0 - OK [INFO]  Detected 2 control-plane nodes [INFO]  Transport: ssh [OK]    STONITH devices found and enabled [OK]    Both nodes ONLINE in Pacemaker [OK]    All daemons healthy on both nodes [OK]    etcd quorum healthy (2/2 voters) [OK]    Fencing secrets correctly bound [INFO]  All non-disruptive checks passed When something fails:
====
----

When non-disruptive checks fail, the output looks something as follows:

[source,terminal]
----

[INFO]
====

>   OpenShift version: 4.20.0 - OK [INFO]  Detected 2 control-plane nodes [INFO]  Transport: ssh [ERROR] No STONITH devices found - fencing is not configured

Reviewer 2: ```suggestion
OpenShift version: 4.20.0 - OK [INFO]  Detected 2 control-plane nodes [INFO]  Transport: ssh [ERROR] No STONITH devices found - fencing is not configured
```
